### PR TITLE
[front] style: smaller top bar on mobile (and more)

### PR DESCRIPTION
**Pull request by GresilleSiffle**
### Description

(1) On desktop, it is now possible to go back to the home page of the current poll by clicking on the Tournesol title.

(2) The Tournesol logo replaces the burger menu icon and opens the side bar, on all devices. I don't know if it will be as user friendly as the burger menu icon.

(3) The top bar is smaller on all devices to saves space for the page content.

(4) The close icon of the search bar now looks like a "go back" icon, instead of a "clear input" to avoid confusion. 

(5) The `presidentielle2022` poll has been "archived". It has been removed from the top bar, and from the home page. All polls are still accessible from the footer.

The component `PollListSection` is still present in the home page of the `presidentielle2022` to ease the access to the main Tournesol poll.

(6) The section title "Statistics" has been added above the statistics section on the home page.

**to-do**
- [x] remove all other links to the poll `presidentielle2022` ?

#### Mobile

| before                                                                                                   | after                                                                                                    |
|----------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|
| cap1 | capture |

#### Desktop

| before | after |
|---|---|
| cap2 | capture |

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
  - if not, delete the related issues section
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [x] The tests pass and have been updated if relevant
- [x] The code quality check pass

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

### CODE REVIEW

File: frontend/src/features/frame/components/topbar/Logo.tsx [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
+          border: '1px solid rgba(0, 0, 0, 0.12)',
+          backgroundColor: 'rgba(29, 26, 20, 0.08)',
+          '&:hover': {
+            backgroundColor: 'rgba(29, 26, 20, 0.16)',
+          },
+        }}
       >
-        <Menu />
+        <img src="/svg/LogoSmall.svg" alt="Tournesol logo" />
       </IconButton>
```
Comment (amatissart) [inline frontend/src/features/frame/components/topbar/Logo.tsx:39]: The border looks good to me. But based on the feedback we got on Discord, I think I would prefer to keep the "burger" to open/close the drawer. And move the Tournesol logo inside the `<InternalLink>`.

File: frontend/src/features/frame/components/topbar/TopBar.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { BeforeInstallPromptEvent } from '../../pwaPrompt';
 
 // Allow to position contents like the footer relatively to the top of the
 // page.
-export const topBarHeight = 80;
+export const topBarHeight = 62;
```
Comment (amatissart) [inline frontend/src/features/frame/components/topbar/TopBar.tsx:18]: I think we also need to update the css override for notifications in "index.css":
https://github.com/tournesol-app/tournesol/blob/b39e6e5f4e6202da70318febd6d6fbaaebd72514/frontend/src/index.css#L37

Comment (GresilleSiffle) [reply frontend/src/features/frame/components/topbar/TopBar.tsx:18]: Fixed by https://github.com/tournesol-app/tournesol/pull/1976/commits/135bea69fa3432baec7182089084890d3a222b6a

File: frontend/src/features/frame/components/topbar/Logo.tsx (hunk 1 above)
Comment (GresilleSiffle) [reply frontend/src/features/frame/components/topbar/Logo.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1976/commits/9df125e52a62321c3f4764dde85f833efb66b9f5

Comment (GresilleSiffle) [reply frontend/src/features/frame/components/topbar/Logo.tsx]: Oops, I created an invalid HTML by adding a `<div>` inside a `<a>`

Comment (GresilleSiffle) [reply frontend/src/features/frame/components/topbar/Logo.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1976/commits/fa322803b71493f639c2744f20fc21f3c9bc113e

**Review by amatissart — APPROVED** [review]

**Pull request merged**